javaslang/src/main/java/javaslang/collection/Arrays2.java

```diff
@@ -13,10 +13,49 @@
  * @author Pap Lőrinc
  * @since 3.0.0
  */
+@SuppressWarnings({"unchecked", "SuspiciousArrayCast"})
 final class Arrays2 { // TODO reuse these in `Array` also
```

Comment:
Our approach is to name them the same and fully qualify the `java.*` stuff. It will not leak to the outside because these helper classes are package-private / internal.